infrastructure/test_config.tf

```diff
@@ -0,0 +1,44 @@
+# configuration for end-to-end tests
+
+resource "azurerm_key_vault_secret" "s2s-url-for-tests" {
+  name      = "s2s-url-for-tests"
+  value     = "${local.s2s_url}"
+  vault_uri = "${module.key-vault.key_vault_uri}"
+}
+
+resource "azurerm_key_vault_secret" "s2s-name-for-tests" {
+  name      = "s2s-name-for-tests"
+  value     = "send_letter_tests"
+  vault_uri = "${module.key-vault.key_vault_uri}"
+}
+
+resource "azurerm_key_vault_secret" "s2s-secret-for-tests" {
+  name      = "s2s-secret-for-tests"
+  value     = "${data.vault_generic_secret.tests_s2s_secret.data["value"]}"
+  vault_uri = "${module.key-vault.key_vault_uri}"
+}
+
+resource "azurerm_key_vault_secret" "idam-url-for-tests" {
+  name      = "idam-url-for-tests"
+  value     = "${var.idam_api_url}"
+  vault_uri = "${module.key-vault.key_vault_uri}"
+}
+
+// TODO: change the client to a dedicated test client (https://tools.hmcts.net/jira/browse/RPE-412)
+resource "azurerm_key_vault_secret" "idam-client-id-for-tests" {
+  name      = "idam-client-id-for-tests"
+  value     = "cmc_citizen"
+  vault_uri = "${module.key-vault.key_vault_uri}"
+}
+
+resource "azurerm_key_vault_secret" "idam-client-secret-for-tests" {
+  name      = "idam-client-secret-for-tests"
+  value     = "${data.vault_generic_secret.test_idam_client_secret.data["value"]}"
+  vault_uri = "${module.key-vault.key_vault_uri}"
+}
+
+resource "azurerm_key_vault_secret" "idam-redirect-uri-for-tests" {
+  name      = "idam-redirect-uri-for-tests"
+  value     = "https://cmc-citizen-frontend-aat-staging.service.core-compute-aat.internal/receiver"
```

Comment:
As aat is harcoded will it work for other environments such as demo

Reply:
Yes, all that matters here is that it's one of the whitelisted URLs. I'll add a comment, though.